### PR TITLE
Create login/authentication plugin

### PR DESCRIPTION
**Pull request by millerman86**
Please accept my humble pull request!!

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Server } from 'hapi';
 import CustomersPlugin from './plugins/customers';
-
+import Authentication from './plugins/authentication';
```
Comment (TylerGarlick) [inline src/index.js]: Rename the plugin to standardize the plugins with a suffix of Plugin `AuthenticationPlugin`

File: src/plugins/authentication.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        return reply(Boom.unauthorized('Login Failed'));
+
+
+      }
+
+    }
+  });
+
+  next();
```
Comment (TylerGarlick) [inline src/plugins/authentication.js]: I would add a return here.  Just a style thing.

File: src/plugins/authentication.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          password: Joi.string().min(7).required()
+        }
+      }
+    },
+    handler: {
+      async: async(request, reply) => {
+        let {email, password} = request.payload;
+        const customers = await collection('customers');
+        const foundCustomer = await customers.findOne({email});
+
```
Comment (TylerGarlick) [inline src/plugins/authentication.js]: Remove some of the additional spacing

File: src/plugins/authentication.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        if (foundCustomer && Bcrypt.compareSync(password, foundCustomer.hashpassword)) {
+          delete foundCustomer.hashpassword;
+          const token = jwt.sign(foundCustomer, "secret", {expiresIn: '1 day' });
+          return reply({token});
+        }
+
```
Comment (TylerGarlick) [inline src/plugins/authentication.js]: Spacing

File: src/plugins/authentication.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+plugin.attributes = {
+  name: 'authentication',
+  version: '1.0.0'
+};
+
+export default plugin;
+
```
Comment (TylerGarlick) [inline src/plugins/authentication.js]: Spacing

**Review by TylerGarlick — CHANGES_REQUESTED** [review]

**Comment by tcrock18** [comment]
👍

**Review by TylerGarlick — APPROVED** [review]

**Pull request merged**